src/winmain.cpp

```diff
@@ -69,6 +72,98 @@ gup [-verbose] [-vVERSION_VALUE] [-pCUSTOM_PARAM]\r\
 
 std::string thirdDoUpdateDlgButtonLabel;
 
+class CUXHelper
+{
+public:
+	CUXHelper()
+	{
+		_hUXTheme = ::LoadLibrary(TEXT("uxtheme.dll"));
+		if (_hUXTheme)
+			_enableThemeDialogTextureFuncAddr = reinterpret_cast<ETDTProc>(::GetProcAddress(_hUXTheme, "EnableThemeDialogTexture"));
+
+		g_hMsgBoxHook = SetWindowsHookEx(WH_CALLWNDPROC, CallWndProc, NULL, GetCurrentThreadId());
+	}
+
+	~CUXHelper()
+	{
+		if (_hUXTheme)
+			::FreeLibrary(_hUXTheme);
+
+		UnhookWindowsHookEx(g_hMsgBoxHook);
+	}
+
+	void goToScreenCenter(HWND hwnd)
+	{
+		RECT screenRc;
+		::SystemParametersInfo(SPI_GETWORKAREA, 0, &screenRc, 0);
+
+		POINT center;
+		center.x = screenRc.left + (screenRc.right - screenRc.left) / 2;
+		center.y = screenRc.top + (screenRc.bottom - screenRc.top) / 2;
+
+		RECT rc;
+		::GetWindowRect(hwnd, &rc);
+		int x = center.x - (rc.right - rc.left) / 2;
+		int y = center.y - (rc.bottom - rc.top) / 2;
+
+		::SetWindowPos(hwnd, HWND_TOP, x, y, rc.right - rc.left, rc.bottom - rc.top, SWP_SHOWWINDOW);
+	}
+
+	void enableDlgTheme(HWND hwnd)
+	{
+		if (_enableThemeDialogTextureFuncAddr)
+		{
+			_enableThemeDialogTextureFuncAddr(hwnd, ETDT_ENABLETAB);
+			redraw(hwnd);
+		}
+	}
+
+	static LRESULT CALLBACK CallWndProc(int nCode, WPARAM wParam, LPARAM lParam)
+	{
+		if (nCode == HC_ACTION)
+		{
+			CWPSTRUCT* pcwp = (CWPSTRUCT*)lParam;
+
+			if (pcwp->message == WM_INITDIALOG)
+			{
+				setIcon(pcwp->hwnd, appIconFile);
+			}
+		}
+
+		return CallNextHookEx(g_hMsgBoxHook, nCode, wParam, lParam);
+	}
+
+	static void setIcon(HWND hwnd, string iconFile)
+	{
+		if (!iconFile.empty())
+		{
+			HICON hIcon = nullptr, hIconSm = nullptr;
+
+			hIcon = reinterpret_cast<HICON>(LoadImageA(NULL, iconFile.c_str(), IMAGE_ICON, 32, 32, LR_LOADFROMFILE));

```

Comment:
Only 1 `hIcon` is enough, if you use LoadIcon for loading the whole icon which contains all the formats (including ICON_BIG & ICON_SMALL, which are not necessarily 16 & 32 respectively)...

Reply:
> https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-loadicona
 LoadIcon can only load an icon whose size conforms to the SM_CXICON and SM_CYICON system metric values. Use the LoadImage function to load icons of other sizes.

IMO, it is good to use `LoadImage`, otherwise client of WinGup will have enforcement of having exact size of icon.